crates/parcel_wasm/src/file_system.rs

```diff
@@ -0,0 +1,97 @@
+use std::collections::HashMap;
+use std::path::Path;
+use std::path::PathBuf;
+
+use dashmap::DashMap;
+use napi::bindgen_prelude::Either3;
+use napi::JsBoolean;
+use napi::JsBuffer;
+use napi::JsFunction;
+use napi::JsString;
+use napi_derive::napi;
+use parcel_resolver::FileSystem;
+
+use crate::function_ref::FunctionRef;
+
+pub struct FileSystemWasm {
```

Comment:
https://github.com/parcel-bundler/parcel/pull/9759/files#diff-02822a9f6bd8b5da0af7e92d669a665c2183ff1e763618817085dc5f9eebd3c6 

was a safe implementation of this